src/DotLiquid/Context.cs

```diff
@@ -437,14 +439,14 @@ private object Variable(string markup, bool notifyNotFound)
 
                 // If object is a KeyValuePair, we treat it a bit differently - we might be rendering
                 // an included template.
-                if (IsKeyValuePair(@object) && (part.Equals(0) || part.Equals("Key")))
+                if (IsKeyValuePair(@object) && (part.Equals(0L) || part.Equals("Key")))

```

Comment:
We need indexes to work on any integer type. Condition class already had type-insensitive equal test capability.